### PR TITLE
fair weights to average in one-time from two-time

### PR DESCRIPTION
**Pull request by ambarb**
* changed `one_time_from_two_time()` to fairly weight pixel mean in diagonal
  * used `np.nanmean()` in case of dropped frames
  * handles multiple two-times 
* updated test to include checking on multiple two-time arrays
  *  cover both major changes to function
  * added one_time calculation to check contrast ratio is unaffected by future change. assumes contrast calculations are already correct for all functions.

In the future, should I branch skbeam for my updates and then fork ?

The tests passed locally so let me know if I need to do anything else before someone is willing to review.

resulting_2time

compare_one_times

### CODE REVIEW

**Comment by ambarb** [comment]
Looks like most of the travis changes to test_correlation.py were other parts of the code that haven't been tested by Travis CI in a while.   

Note PR motivated by https://github.com/scikit-beam/scikit-beam/issues/565

File: skbeam/core/tests/test_correlation.py [hunk 1]
```diff
@@ -181,69 +190,95 @@ def test_auto_corr_scat_factor():
...
 
 def test_one_time_from_two_time():
+    np.random.seed(333)
     num_lev = 1
     num_buf = 10  # must be even
     x_dim = 10
     y_dim = 10
     stack = 10
     imgs = np.random.randint(1, 3, (stack, x_dim, y_dim))
+    for i in range(imgs.shape[0]):
```
Comment (dmgav) [inline skbeam/core/tests/test_correlation.py:221]: I think this could be accomplished as `imgs[:, 0, 0] += 5`

Comment (ambarb) [reply skbeam/core/tests/test_correlation.py:221]: ok.  will work on that.

Comment (ambarb) [reply skbeam/core/tests/test_correlation.py]: @dmgav your suggestion is better. thanks for the kind review.
 Just a general git etiquette question:  once the tests pass, do I click resolve conversation or do one of you?

**Comment by ambarb** [comment]
@tacaswell @dmgav we noticed almost 2 weeks ago flake8 is configured in two files `setup.cfg` and `.flake8`.  Should we combine?  I can do it here if you say which way you want this done.

File: skbeam/core/cdi.py [hunk 2]
```diff
@@ -307,21 +322,24 @@ def cdi_recon(diffracted_pattern, sample_obj, sup,
...
-    if pi_modulus_flag == 'real':
+    if pi_modulus_flag == "real":
         real_operation = True
-    elif pi_modulus_flag == 'complex':
+    elif pi_modulus_flag == "complex":
         real_operation = False
     else:
-        raise ValueError('py_modulus_flag must be one of {"complex",'
-                         'real"} not' '{!r}'.format(pi_modulus_flag))
+        raise ValueError(
```
Comment (dmgav) [inline skbeam/core/cdi.py:335]: `        raise ValueError('py_modulus_flag must be one of ("complex",'
                         '"real") not' '{!r}'.format(pi_modulus_flag))
`
I think that this is the issue with flake8. The latest version of flake8 seems to make more comprehensive code analysis, so it finds issues in all projects. This is an example when flake8 found a real bug: the code wouldn't work as expected if this branch was hit.

Comment (dmgav) [reply skbeam/core/cdi.py:335]: (The difference is that {} is replaced by ())

Comment (dmgav) [reply skbeam/core/cdi.py:335]:
```        
            raise ValueError('py_modulus_flag must be one of ("complex", "real")' 
                 'not {!r}'.format(pi_modulus_flag))
```
Better formatted version

Comment (ambarb) [reply skbeam/core/cdi.py:335]: Ahh.  That makes sense, I was just focused on why it thought things were not in the required format.  I was going to go back in time and start fresh so your message is timely.  

I don't understand why the files I haven't touched were getting flagged.  Do the travis checks check everything?

**Comment by dmgav** [comment]
> @tacaswell @dmgav we noticed almost 2 weeks ago flake8 is configured in two files `setup.cfg` and `.flake8`. Should we combine? I can do it here if you say which way you want this done.

I think that the best way is to put configuration for flake8 only in `.flake8`, since this configuration file is used when you run flake8 both locally and on CI. I am not sure if configuration from `setup.cfg` is used at all, but some settings there seem to be out of date. For example, I don't think that the file `xrf_model.py` should be skipped. Currently the settings in `.flake8` are sufficient for the test to pass, so I think that flake8 entry can be safely removed from `setup.cfg`.

File: skbeam/core/cdi.py (hunk 2 above)
Comment (dmgav) [reply skbeam/core/cdi.py:335]: flake8 check the whole project every time you run it, unless you specify the files you want to check. You may have older version of flake8 installed locally, so you may need to update it to see the same result as on Travis, which is always using the latest version.

**Comment by dmgav** [comment]
I think that 'flake8' could be simply removed from `setup.cfg` without adding anything to `.flake8`, because the test is passed successfully on the whole project with configuration already contained in `.flake8`. We want to keep the number of excluded files as low as possible.

**Comment by ambarb** [comment]
ok.  removed entire line and all tests passed. will push that version of .flake8

**Comment by ambarb** [comment]
> ok. removed entire line and all tests passed. will push that version of .flake8

**Comment by ambarb** [comment]
I am not sure how I managed to close.

**Pull request merged**